src/objective_types.jl

```diff
@@ -228,18 +228,18 @@ function value!(obj, x)
 end
 
 
-function _unchecked_grad!(obj, x)
+function _unchecked_gradient!(obj, x)
     obj.g_calls .+= 1
     copy!(obj.last_x_g, x)
-    obj.g!(x)
+    obj.g!(x, stor)
```

Comment:
There is no `stor` argument here, do you mean `obj.g`? Alternatively we'd have to add a `storage` parameter to `gradient!` and `value_gradient!` etc. as well.